tree_test.go

```diff
@@ -0,0 +1,183 @@
+package rsmt2d
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegisterTree(t *testing.T) {
```

Comment:
[Optional] I typically recommend including a descriptive summary for test functions to enhance readability and facilitate quicker comprehension of the test's purpose and results.

Reply:
gotcha i will add sumary of the test functions